File: examples/go/user/read/readAccountByUsernameOrEmail.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/user"
)

// Read an account by id, username or email. Only one need to be specified.
func ReadAccountByUsernameOrEmail() {
	userService := user.NewUserService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := userService.Read(&user.ReadRequest{
		Username: "usrname-1",
	})
	fmt.Println(rsp, err)
}

File: examples/js/user/read/readAccountByUsernameOrEmail.js
const { UserService } = require("m3o/user");

// Read an account by id, username or email. Only one need to be specified.
async function readAccountByUsernameOrEmail() {
  let userService = new UserService(process.env.MICRO_API_TOKEN);
  let rsp = await userService.read({
    username: "usrname-1",
  });
  console.log(rsp);
}

readAccountByUsernameOrEmail();
